core/lib/zksync_core/src/db_pruner/mod.rs

```diff
@@ -0,0 +1,499 @@
+mod metrics;
+pub mod prune_conditions;
+
+use std::{fmt, sync::Arc, time::Duration};
+
+use async_trait::async_trait;
+use tokio::sync::watch;
+use zksync_dal::ConnectionPool;
+use zksync_types::L1BatchNumber;
+
+use crate::db_pruner::metrics::{MetricPruneType, METRICS};
+
+#[derive(Debug)]
+pub struct DbPrunerConfig {
+    pub soft_and_hard_pruning_time_delta: Duration,
+    pub next_iterations_delay: Duration,
```

Comment:
- Bikeshedding: I'd call this field something like `soft_pruning_check_interval`; the current name is quite vague.
- It makes sense to document `DbPrunerConfig` fields so that their semantics is clearer. These docs could then be copied to the EN operation manual etc.

Reply:
Those config values are imo generally not something to tinker with by EN operators as their meaning is very subtle

Reply:
I'll add documentation though